src/frontend/Komponenter/Behandling/Totrinnskontroll/SendTilBeslutterFooter.tsx

```diff
@@ -66,6 +88,34 @@ const SendTilBeslutterFooter: React.FC<{
             .finally(() => settLaster(false));
     };
 
+    const checkBoxEndring = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSkalOppretteFremleggsoppgave(event.target.checked);
+    };
+
+    useEffect(() => {
```

Comment:
Lurer på om bruk av useEffect på dette viset er riktig ? (Ønsker å utføre kallene i useEffect ved instansiering av komponenten)

Reply:
Ja, men jeg tenker den kun trengs å gjøres hvis behandlingen er førstegangsbehandling? 
Sånn at den då burde bli
```js
useEffect(() => {
if(behandling.type === Behandlingstype.FØRSTEGANGSBEHANDLING) {
  ...
}
}, [axios, behandling])
```

Reply:
Hadde nok blitt mer riktig (og i hvertfall stått mer i stil til resten av kodebasen vår) om du lagde en egen hook for fremleggsoppgaver, typ `useFremmleggsoppgave`. Her legger du inn alle kall mot backend. Du burde også benytte  `<Dataviewer/>` komponenten slik at komponentene som er avhengige av verdiene som hentes ikke vises før verdiene er klare. 

Spør gjerne om hjelp med dette, da det krever en god del omskriving

Reply:
Er det noen grunn til at vi må ha en egen hook for fremleggsoppgaver fremfor å bruke useEffect ? Trenger vi Dataviewer i dette tilfellet ?

Reply:
@blommish Vi vil sette kanOppretteFremleggsoppgave til false i ef-sak hvis det ikke er en førstegangsbehandling. Du tenker at vi bør fiske frem behandlingstype her, og så sjekke her også for å unngå et ekstra kall ?

Reply:
Ja, men har det noe å si at vi utfører det kallet vs å fiske ut typen ?

Reply:
Synes det er ryddig å ikke gjøre unødvendige kall
Tenker vi egentlige ikke trenger en egen `hook` hvis Fremleggsoppgave trekkes ut i en egen komponent

Reply:
Vi hentet ut typen og sjekker 👍